sdks/python/apache_beam/ml/inference/tensorflow_inference.py

```diff
@@ -0,0 +1,245 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# pytype: skip-file
+
+import enum
+from typing import Any, Union
+from typing import Callable
+from typing import Dict
+from typing import Iterable
+from typing import Optional
+from typing import Sequence
+
+import sys
+import numpy
+import tensorflow as tf
+
+from apache_beam.ml.inference import utils
+from apache_beam.ml.inference.base import ModelHandler
+from apache_beam.ml.inference.base import PredictionResult
+
+__all__ = [
+    'TFModelHandlerNumpy',
+    'TFModelHandlerTensor',
+]
+
+TensorInferenceFn = Callable[[
+    tf.Module,
+    Sequence[Union[numpy.ndarray, tf.Tensor]],
+    Optional[Dict[str, Any]],
+    Optional[str]
+],
+                             Iterable[PredictionResult]]
+
+
+class ModelType(enum.Enum):
+  """Defines how a model file should be loaded."""
+  SAVED_MODEL = 1
+
+
+def _load_model(model_uri, model_type):
+  if model_type == ModelType.SAVED_MODEL:
+    return tf.keras.models.load_model(model_uri)
```

Comment:
From https://docs.google.com/document/d/1c2rWX7fA7UAl2qabzEXg5r7_zwosHfGo6mg1gz8VdSQ/edit?pli=1&disco=AAAAoHqF1dw I thought we were going to use `TFHub.resolve` - is there a reason we're switching back to load_model?

Reply:
Loading models saved with `model.save()/tf.saved_model.save()` is throwing an input type exception that input is of type tf.Tensor but required a Tensorspec input. I'm looking out if we need to save the model a particular way for it to be loadable via `hub.load()`

Reply:
CC: @jrmccluskey

Reply:
Hmm. The tf.Tensor input type feels out of left field. `hub.load()` is roughyl equivalent to calling `resolve()` and then `tf.saved_model.load()` on the result, not sure where there's an opportunity for a type mismatch like a tensor sneaking in there

Reply:
@riteshghorse can you give us a small example to reproduce? I can try to debug as well.

Reply:
I tried and it run successfully. Can you share the model you used to save?
```
def create_model():
  model = tf.keras.Sequential([
    keras.layers.Dense(512, activation='relu', input_shape=(784,)),
    keras.layers.Dropout(0.2),
    keras.layers.Dense(10)
  ])
  return model

model = create_model()
model.save('/tmp/model.h5')
m1 = tf.keras.models.load_model('/tmp/model.h5')
print(m1)
# <keras.engine.sequential.Sequential at 0x1507a6910>
```

Reply:
@AnandInguva load_model works fine. I'm trying with `tensorflow_hub.load()`

Reply:
Okay got it, so I tried two variants when the model is saved using `model.save()`
1. `tf.saved_model.load(hub.resolve(model_uri))`
2. `tf.keras.models.load_model(hub.resolve(model_uri))`

First one throws error mentioned above and the second worked fine

Reply:
Does the pattern of calling hub.resolve() on the path then using the tf.keras.models.load_model() function on the resolved path work?

Reply:
Ah I asked that while you were typing out your comment. Interesting. I'm not familiar enough with tf to know if there's any mechanical difference between tf core load and keras load, but it seems like we're hitting a snag somewhere in the former

Reply:
yeah.. those two are different APIs to load the model it seems based on this: https://stackoverflow.com/a/74617677/7042941

Reply:
I tested this by loading a model from tensorflow hub and performing a prediction in a notebook. I'll add that example to this PR itself.